### PR TITLE
In memory impl

### PR DESCRIPTION
**Pull request by krukru**
This is an implementation for the InMemory storage adapter, it is basically a rework of the APC adapter. 

I found it useful for gauges combined with the pushgateway when I don't really need to persist the data.

### CODE REVIEW

**Comment by bracki** [comment]
Nice. I'll give it a try. Seems to pass the tests. I'll get back to it next week.

**Comment by bracki** [comment]
@krukru Do you mind adding an example to the README?

File: src/Prometheus/Storage/InMemory.php [hunk 1]
```diff
@@ -3,42 +3,132 @@
...
+                'help' => $metaData['help'],
+                'type' => $metaData['type'],
+                'labelNames' => $metaData['labelNames'],
+            );
+            foreach ($metric['samples'] as $key => $value) {
+                $parts = explode(':', $key);
+                $labelValues = $parts[2];
+                $data['samples'][] = array(
+                    'name' => $metaData['name'],
+                    'labelNames' => array(),
```
Comment (bweston92) [inline src/Prometheus/Storage/InMemory.php]: Line 12 you're constructing arrays differently. Maybe keep some consistency?

**Comment by krukru** [comment]
Agreed. I'll do some cleanup and add instructions to the readme soon

**Comment by krukru** [comment]
Will this be ok?

**Pull request merged**